### ISSUE TITLE
CS8600: possible null assignment false positive when 'is true' is in the expression

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#57382]
**Version Used**: VS 2019 Update 11

**Steps to Reproduce**:

[Sharplab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQygg5hG6CeADgJYB2WAPgMSlwA2dUwdEABBKUywLABQfAAQAMrAQEYALAG4+fAG6ZWAE2IBjGKwC8rUhADurACJqYxAPacE+ADzihAGlFihAPgAUAShm9iYVm5V1ADoAFSsAcVwANSg6OAg3ACIAawh8RMczOA07AH5WBTiID1ZiZFYYBHiPPgBvPlZGpxE5LQLY+O8AXz4gA===)
```cs
#nullable enable

using System.Collections.Generic;

var dict = new Dictionary<string, string>();
if (dict.TryGetValue("key", out string? value) is true)
{
    string v = value;
}
```

**Expected Behavior**:

No warnings.

**Actual Behavior**:

A warning on the `string v` assignment:

> warning CS8600: Converting null literal or possible null value to non-nullable type.
